controllers/hvpa_controller.go

```diff
@@ -1668,11 +1730,21 @@ func (r *HvpaReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	}
 
 	podSource := source.Kind{Type: &corev1.Pod{}}
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&autoscalingv1alpha1.Hvpa{}).
-		Owns(&autoscaling.HorizontalPodAutoscaler{}).
-		Owns(&vpa_api.VerticalPodAutoscaler{}).
-		Watches(&podSource, podEventHandler).
-		WithEventFilter(pred).
-		Complete(r)
+	if !r.IsAutoscalingV2Enabled {
```

Comment:
Nit: Code can be reduced here by branching just the HPA object instantiation instead of the controller creation call.